### PR TITLE
#18 Initialize New Footer Component

### LINKED ISSUE #18: Create Reusable Footer Component
**Description**
Create the footer reusable component that will be shown on the bottom of all the main pages of the CIMUN website

**UC**
* As a user, I want to see important info/links on the bottom of my current page in order to efficiently get my questions answered without having to figure out which other page might have what I am looking for and to see how to get in contact with the CIMUN team

**AC**
* This component should have the powered by Vercel tag
* the component should have CIMUN logo
* the component should have contact info (e.g. href="mailto:SC@cimun.org")
* the component should have links to social media accounts
* the component should have relevant links to other sites like merch
* the component should have link to MIMUN website

### PR DESCRIPTION
**Pull request by FB-Reda**
* Add FontAwesome
* Add new footer and associated storybook/styles
* Improved storybook setup
* basically transferred over the new header so far.  Improvements to follow.
* closes #18

### CODE REVIEW

**Review by sdmungov — COMMENTED** [review]
I think my pr merging caused a couple conflicts so gotta resolve those but they look to be pretty small. 

Also did you check to see if this is responsive and mobile friendly, I think the only necessary change there looks to be making it one single column with things centered instead of left aligned after a mobile breakpoint like 650, below is what it currently looks like on a mobile device
Screen Shot 2021-06-29 at 11 57 40 PM

Other than that things look good

**Comment by FB-Reda** [comment]
@he-james can we color the vercel logo white in the footer or does that break guidelines?

**Comment by he-james** [comment]
@FB-Reda see branding guidelines here. https://vercel.com/design#dark-logotype we can use a white logo or the powered by banners

File: components/footer/Footer.styles.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import styled from 'styled-components'
+import { colors } from '../../styles/colors';
+
+export const StyledFooter = styled.footer`
+  -moz-osx-font-smoothing: grayscale;
+  -webkit-font-smoothing: auto;
+  ;
```
Comment (he-james) [inline components/footer/Footer.styles.ts]: ??

File: components/footer/Footer.styles.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  background-color: ${colors.primaryBlue};
+  border-top: 1px solid #eaeaea;
+  color: white;
+  font-size: 12px;
+  // padding: 50px 0;
```
Comment (he-james) [inline components/footer/Footer.styles.ts]: remove

File: pages/_app.js [hunk 3]
```diff
@@ -1,4 +1,10 @@
-import '../styles/globals.css'
+import '../styles/globals.css';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { fab } from '@fortawesome/free-brands-svg-icons';
+import { fas } from '@fortawesome/free-solid-svg-icons';
+import { far } from '@fortawesome/free-regular-svg-icons'
+
+library.add(fab, fas, far);
```
Comment (he-james) [inline pages/_app.js:7]: gonna assume this is right and pretend i know what this does at all

File: pages/index.tsx [hunk 4]
```diff
@@ -9,7 +10,7 @@ export default function Home() {
   return (
     <Wrapper>
       <Head>
-        <title>Create Next App</title>
+        <title>Chicago International Model UN</title>
         <meta name="description" content="Generated by create next app" />
```
Comment (he-james) [inline pages/index.tsx:14]: eventually this should be removed lolll

File: styles/containers.ts [hunk 5]
```diff
@@ -29,28 +28,28 @@ export const Wrapper = styled.div<ContainerType>`
 
 export const ComponentWrapper = styled.div<ContainerType>`
   margin: ${(props: ContainerType) =>
-      props.margins ? props.margins : "3.9375vw auto"};
+    props.margins ? props.margins : "3.9375vw auto"};
```
Comment (he-james) [inline styles/containers.ts:34]: 4 space vs 2 space war

**Comment by he-james** [comment]
@FB-Reda only nit pick aside from the comments I made is that the headers and the logo should be aligned as needed - could not fix it myself because I am a lowly API developer. Also determine what refinements can be made to improve mobile compatibility per @sdmungov 's comment

image

**Review by he-james — APPROVED** [review]

**Pull request merged**